nservicebus/concept-overview.md

```diff
@@ -11,53 +11,67 @@ A high level overview of the concepts, features and vernacular of NServiceBus.
+
+### Endpoint Instance
+
+An Endpoint Instance is a run-time object that allows to interact with the bus. Endpoint Instances are able to send and receive messages (an exception is a *send-only* endpoint). It runs associated Message Handlers and Sagas to process incoming messages. An Endpoint Instance has a single Input Queue (which can be shared with other instances of the same Endpoint in a scale-out scenario). 
+
+### [Hosting](/nservicebus/hosting)
+
+The act of Hosting refers to running an Endpoint Instance in some process. This can be any .NET process such as a console application, a website, or a Windows service. Multiple Endpoint instances can be hosted in a single process (custom AppDomains are required to achieve this in version 4 and below).
 
-Extra information about a message is communicated over the transport in a serialized collection of key value pairs in a similar way to how http headers are used.
+### [Transport](/nservicebus/transports/)
+
+The Transport is the mechanism that NServiceBus Endpoint Instances use to communicate with each other. NServiceBus supports many different transports in a pluggable manner. For Endpoint Instances to communicate they need to share a common Transport technology.
 
+### Input Queue
+
+Each endpoint instance is assigned a single Input Queue. This queue can be shared among instances of same Endpoint but never with other Endpoints.
 
 ### [Publish Subscribe](/nservicebus/messaging/publish-subscribe)
 
 Publish Subscribe is the interaction of 
 
- * Registering interest in being notified about an Event.
- * That Event being delivered to the endpoint that registered
-
-A "Subscriber" lets the "Publisher" know they're interested in an Event, and the "Publisher" stores the "Subscriber" address so that it knows where to send the Event. 
+ * Registering interest in being notified about an event (*subscriber*).
+ * That event being delivered to the endpoint that registered itself (*publisher*)
 
+Both *publisher* and *subscriber* are Endpoints which means they are logical rather than physical (e.g. `Sales` or `Shipping`, not `Cache-1` or `Web-1`). Each event is delivered to a single Instance of each *subscriber* Endpoint.
 
 ### [Handler](/nservicebus/handlers/)
 
-A Handler is code that processes a message. They are stateless and constructed on a per-message basis. 
+A Message Handler (or simply Handler) is a piece of code that processes a message of a given type. Handlers are not guaranteed to retain any state between processing of messages.
 
 
 ### [Saga](/nservicebus/sagas/)
 
-A Saga can be thought of as a long running Handler that handles multiple Messages and shared state. It is the NServiceBus equivalent of a workflow.
+A saga can be thought of as a long running Handler that handles multiple Messages and shared state. It is the NServiceBus equivalent of a Process Manager pattern.
```

Comment:
We should link to a resource explaining that pattern since most readers won't really know about that pattern (but everyone knows what a workflow is).